apps/meteor/app/e2e/client/rocketchat.e2e.ts

```diff
@@ -437,6 +437,30 @@ class E2E extends Emitter {
 		return decryptedMessageWithAttachments;
 	}
 
+	async decryptPinnedMessage(message: IMessage) {
+		if (!message?.attachments || !message.attachments[0]?.text) {
+			return message;
+		}
+
+		const e2eRoom = await this.getInstanceByRoomId(message.rid);
+		const pinnedMessage = message.attachments[0]?.text;
+
+		if (!e2eRoom || !pinnedMessage) {
+			return message;
+		}
+
+		const data = await e2eRoom.decrypt(pinnedMessage);
+
+		if (!data) {
+			return message;
+		}
+
+		const decryptedPinnedMessage = { ...message } as IMessage & { attachments: MessageAttachment[] };
+		decryptedPinnedMessage.attachments[0].text = data.text;
```

Comment:
Does the message have more than one attachment? Otherwise u can assign this on the same assignment above

Reply:
Probably not, but just like to be more explicit, better code readability.